api_core/google/api_core/bidi.py

```diff
@@ -574,11 +581,12 @@ def stop(self):
                 self._thread.join()
 
             self._thread = None
+            self._ready.clear()
 
     @property
     def is_active(self):
         """bool: True if the background thread is active."""
-        return self._thread is not None and self._thread.is_alive()
+        return self._thread is not None and self._ready.is_set() and self._thread.is_alive()
```

Comment:
This is redundant.  If ready is set, you know that the thread callback has already started to run, since _ready.set() is the first line of that callback.

Reply:
I had original removed `is_alive` since it shouldn't be possible for ready to be set and alive not to be. I can remove it again to clarify what is going on.